flatten.go

```diff
@@ -0,0 +1,167 @@
+package authzed
+
+import (
+	"fmt"
+
+	api "github.com/authzed/authzed-go/arrakisapi/api"
+)
+
+// FlattenExpand reduces an ExpandResponse into the slice of Users present in
+// the expansion.
+//
+// Notably, this removes the context of which relations caused users to be
+// included in the expansion, but often you only need to know which users are
+// present.
+func FlattenExpand(resp *api.ExpandResponse) []*api.User { return flatten(resp.TreeNode) }
+
+func flatten(node *api.RelationTupleTreeNode) []*api.User {
+	switch typed := node.NodeType.(type) {
+	case *api.RelationTupleTreeNode_IntermediateNode:
+		switch typed.IntermediateNode.Operation {
+		case api.SetOperationUserset_UNION:
+			return flattenUnion(typed.IntermediateNode.ChildNodes)
+		case api.SetOperationUserset_INTERSECTION:
+			return flattenIntersection(typed.IntermediateNode.ChildNodes)
+		case api.SetOperationUserset_EXCLUSION:
+			return flattenExclusion(typed.IntermediateNode.ChildNodes)
+		}
+	case *api.RelationTupleTreeNode_LeafNode:
+		users := newUserSet()
```

Comment:
Maybe have a `.extend` that takes in a slice?

Edit: or just pass `typed.LeafNode.Users...` to `add`?